src/core/mormot.core.interfaces.pas

```diff
@@ -5391,6 +5392,14 @@ destructor TInterfaceResolverInjected.Destroy;
   end;
 end;
 
+procedure TInterfaceResolverInjected.DeleteResolver(aResolver: TInterfaceResolver);
+var
+  ix: integer;
+begin
+  ix := ObjArrayFind(fResolvers,aResolver);
+  if ix >= 0 then
+    ObjArrayDelete(fResolvers,ix);

```

Comment:
I guess you should also check within fResolversToBeReleased[] to avoid potential GPF in Destroy

Reply:
BTW I guess you could just write ObjArrayDelete(fResolvers, aResolver); with no intermediate ix variable

Reply:
Done.